### DISCUSSION TITLE
error in Compile libMesh for HPC Cluster installation

### DISCUSSION THREAD

**Discussion by Amir1361** [idaholab/moose#21342] [Q&A Getting Started] [answered]
Hi there,

I got the following error in moose installation. I have attached the terminal outputs. Could you please help me to solve the issue?
[output.txt](https://github.com/idaholab/moose/files/8931179/output.txt)

Thank you very much!

Amir

**Comment by GiudGiud** [comment, marked as answer]
Hello

Seems you are missing some libraries to install libmesh
In particular look at libtirpc-devel, see if it is installed on your system

Guillaume

**Reply by GiudGiud** [reply]
see this post for more details
https://github.com/idaholab/moose/discussions/20479

**Reply by Amir1361** [reply]
Thanks Guillaume, I have checked the #20479, unfortunately, it does not work for me yet. The libtirpc-devel version is 1.1.4.
[output_libtirpc_devel.txt](https://github.com/idaholab/moose/files/8942533/output_libtirpc_devel.txt). Thank you for your help!

**Reply by GiudGiud** [reply]
well even if you do have it installed, libmesh is not finding it. 
Please have a look at Roy's answer in that thread to help it find the file
https://github.com/idaholab/moose/discussions/20479#discussioncomment-2311626

**Reply by Amir1361** [reply]
I found the rpc.h/xdr.h in the "/usr/include/tirpc" path. How can I address it for limbash? 
the following comand doesn't work:
$./scripts/update_and_rebuild_libmesh.sh configure CPPFLAGS=/usr/include/tirpc/rpc/

**Reply by GiudGiud** [reply]
does CPPFLAGS need to be passed before the command?

**Reply by GiudGiud** [reply]
also you do not want to erase CPPFLAGS and set it to this value, you only want to amend it so:
CPPFLAGS=/usr/include/tirpc/rpc/:$CPPFLAGS

**Reply by mlesueur** [reply]
Hi @GiudGiud @Amir1361 ,
I have the same error while trying to compile libmesh on a cluster where I have to load libtirpc externally via a module. I can see that the files `rpc/rpc.h` and `rpc/xdr.h` are located in `$LIBTIRPC_MODULE_PATH/include/tirpc`. 
The command I am trying to run is:
`./scripts/update_and_rebuild_libmesh.sh CPPFLAGS=-I/$LIBTIRPC_MODULE_PATH/include/tirpc
`
But the error is still:
```
checking for XDR support in /usr/include/tirpc... no
configure: error: *** XDR was not found, but --enable-xdr-required was specified.
```
Did you manage to solve your problem @Amir1361 ?
Is there a mistake I am making? 
Cheers,

 Martin

**Reply by GiudGiud** [reply]
Hello you neee to install libtirpc-devel on your system

is this a computing cluster or your own machine?

**Reply by mlesueur** [reply]
Thank you for your answer @GiudGiud, it's a computing cluster.
Is the libtirpc-devel a different library than libtirpc? Would you mind giving me more details on how to install that then?

**Reply by GiudGiud** [reply]
If it s a computing cluster, the easiest is to ask your cluster administrators to install it

**Comment by Amir1361** [comment]
Hello,

Here are the outputs for Compile Limbash and Compile & Test Moose. It seems that I do not have "Python.h". Could you please help me with this issue? Thanks,

Amir
[install.txt](https://github.com/idaholab/moose/files/9035541/install.txt)
[test.txt](https://github.com/idaholab/moose/files/9035542/test.txt)

**Reply by GiudGiud** [reply]
Missing python.h usually means something is wrong with your python, like you only have an old python 2.7 for example. 

What do the following return:
```
echo $CXX
which mpicc
which python
python --version
```

Guillaume

**Reply by Amir1361** [reply]
Hi Guillaume,

Here are the outputs. Thank you for your help!

![jetstream](https://user-images.githubusercontent.com/55258450/177673556-78b59cd9-7a42-440b-95ca-cd4e7719b749.JPG)

**Reply by milljm** [reply]
Missing python.h means `python-devel` packages are not installed (when using system Python such as /usr/bin/).

Edit: Actually this pretty much sums it up: https://stackoverflow.com/questions/21530577/fatal-error-python-h-no-such-file-or-directory

**Reply by Amir1361** [reply]
@milljm and @GiudGiud Thank you very much!
